test/extractor/ValueSetComponentRuleExtractor.test.ts

```diff
@@ -0,0 +1,103 @@
+import { fshtypes } from 'fsh-sushi';
+import { ProcessableValueSetComponent } from '../../src/processor';
+import { ValueSetConceptComponentRuleExtractor } from '../../src/extractor';
+
+const { FshCode } = fshtypes;
+
+describe('ValueSetComponentRuleExtractor', () => {
+  it('should extract a ValueSetConceptComponentRule from a component with a system and at least one concept', () => {
+    const input: ProcessableValueSetComponent = {
+      system: 'http://example.org/zoo',
+      concept: [
+        {
+          code: 'BEAR'
+        },
+        {
+          code: 'LION',
+          display: 'Lion'
+        }
+      ]
+    };
+    const result = ValueSetConceptComponentRuleExtractor.process(input, true);
+
+    expect(result.inclusion).toBe(true);
+    expect(result.from).toEqual({
+      system: 'http://example.org/zoo'
+    });
+    expect(result.concepts).toHaveLength(2);
+    expect(result.concepts).toContainEqual(new FshCode('BEAR', 'http://example.org/zoo'));
+    expect(result.concepts).toContainEqual(new FshCode('LION', 'http://example.org/zoo', 'Lion'));
+  });
+
+  it('should extract a ValueSetConceptComponentRule from a component with a system, at least one value set, and at least one concept', () => {
```

Comment:
Since FHIR allows for `valueSet` _without_ `system` should we also have one test to demonstrate that scenario?

Reply:
This is something I considered, but intentionally left out. A concept component must have at least one concept, and when there is at least one concept, `system` is required. So, this scenario would be represented as a filter component. So, either `ValueSetProcessor.process` needs some logic to determine which rule extractor to use for a given component, or the extractors need some logic to determine when they've given a component that is valid, but not their responsibility.

Reply:
Spot on.  I forgot that presence of a `concept` requires `system`.  Good call.